### PR TITLE
Proper test cleanup and resolve EOF Error

### PR DESCRIPTION
**Pull request by thodkatz**
As I am trying to migrate to github actions (https://github.com/ilastik/tiktorch/pull/217), I have realized two issues when running the tests:

- Tests don't terminate when one of them fails (pytest is hanging)
- Even when the tests pass, we are getting EOF Errors.

I have attempt to resolve these issue with their respective commits :)

### CODE REVIEW

**Comment by thodkatz** [comment]
CI is failing because it gets a cached environment with these packages installed
```
bioimageio-core           0.6.7                    pypi_0    pypi
bioimageio-spec           0.5.3.post4              pypi_0    pypi
```

, maybe coming from the PR of updating the submodules to v5.

File: Makefile [hunk 1]
```diff
@@ -21,18 +10,23 @@ protos:
...
-	conda run -n $(TIKTORCH_ENV_NAME) pip install . ./vendor/core-bioimage-io-python ./vendor/spec-bioimage-io
+	make install_submodules
 
 run_server:
 	. $$(conda info --base)/etc/profile.d/conda.sh; conda activate $(TIKTORCH_ENV_NAME); python -m tiktorch.server
 
+install_submodules:
+	@echo "Installing submodules $(SUBMODULES)"
+	@for package in $(SUBMODULES) ; do \
+		echo $$package ; \
```
Comment (thodkatz) [inline Makefile:24]: to escape shell variables such as $package in makefiles, we need double $$ :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -63,15 +59,15 @@ jobs:
...
+      - image: condaforge/miniforge3
     working_directory: ~/repo
     steps:
       - restore_cache:
           keys:
             - v1-repo-{{ .Environment.CIRCLE_SHA1 }}
       - restore_cache:
           keys:
-            - v11-dependencies-{{ checksum "environment.yml" }}
+            - v11-dependencies-{{ .Environment.CIRCLE_SHA1 }}
```
Comment (thodkatz) [inline .circleci/config.yml:70]: the environment should be build, whenever there is a change in the `environment.yml` or the submodules. For now just make every run isolated, building the environment from scratch. We will address this issue at https://github.com/ilastik/tiktorch/pull/217

File: Makefile (hunk 1 above)
Comment (k-dominik) [reply Makefile:24]: lovely, thx for checking!

File: tiktorch/server/device_pool.py [hunk 3]
```diff
@@ -71,10 +71,22 @@ def cuda_version(self) -> Optional[str]:
...
+        List available devices on server
+        """
+        return [device for device in self.list_devices() if device.status == DeviceStatus.AVAILABLE]
+
+    def list_reserved_devices(self) -> List[IDevice]:
+        """
+        List reserved devices on server
+        """
+        return [device for device in self.list_devices() if device.status == DeviceStatus.IN_USE]
+
```
Comment (k-dominik) [inline tiktorch/server/device_pool.py:89]: I can see the value of adding these, but as far as I can tell they are not used. The commit message doesn't really explain why these have been added. So no context other than "nice to have" as far as I can see.

File: tiktorch/rpc/mp.py [hunk 4]
```diff
@@ -373,3 +371,7 @@ def listen(self):
                 break
             except Exception:
                 self.logger.error("Error in main loop", exc_info=1)
+
+        self._start_result_sender.join()
+        self._conn.close()
+        self.logger.debug("Closing connection")
```
Comment (k-dominik) [inline tiktorch/rpc/mp.py:377]: great that you fixed that!!

**Review by k-dominik — APPROVED** [review]
Looking great, thank you for fixing this. I also appreciate the renamed variables, makes it easier to follow :)

Also really great that you added the proper cleanup in the rpc!

File: tiktorch/server/device_pool.py (hunk 3 above)
Comment (thodkatz) [reply tiktorch/server/device_pool.py:89]: it is being used here https://github.com/ilastik/tiktorch/pull/218/files#diff-31a22ff2b156f62383aa4c24cb4bf40cd11cf238819548c8c29dc51e7dd0d4a0R63, I wanted to assert that there aren't any reserved devices. Then we would have a function for getting all the devices, and one to get the reserved one. For consistency, I have added to be able to get all the available ones as well, but it is true isn't used. I could remove it.

**Pull request merged**